commons-vfs2/src/main/java/org/apache/commons/vfs2/impl/DefaultFileSystemManager.java

```diff
@@ -208,6 +208,20 @@ public void addProvider(final String[] urlSchemes, final FileProvider provider)
         }
     }
 
+    /**
+     * Unregisters a file system provider.
+     *
+     * @param urlScheme The scheme of the provider.
+     * @since 2.8.0
+     */
+    public void removeProvider(final String urlScheme) {
+        final FileProvider provider = providers.remove(urlScheme);
+        // check whether the same instance is not used somewhere else
+        if (provider != null && providers.values().stream().noneMatch(p -> p == provider)) {
+            closeComponent(provider);
```

Comment:
Hi @boris-petrov 
Thank you for adding a unit test.
Are you saying here that the same provider can be registered in the providers map more than once but under different keys?
If so, you need to test that case as well.